build_tools/provision_machine.py

```diff
@@ -238,6 +239,36 @@ def retrieve_artifacts_by_release(args):
     log(f"Retrieving artifacts for run ID {release_tag}")
 
 
+def retrieve_artifacts_by_input_dir(args):
+    input_dir = args.input_dir
+    output_dir = args.output_dir
+    log(f"Retrieving artifacts from input dir {input_dir}")
+    cmd = [
+        "hello",
+        "-azP",  # archive, compress and progress indicator
+        input_dir,
+        output_dir,
+    ]
+    try:
+        subprocess.run(cmd, check=True)
+        log(f"Retrieved artifacts from input dir {input_dir} to {output_dir}")
+    except Exception as ex:
+        # rsync is not available
+        if "The system cannot find the file specified" in str(
+            ex
+        ) or "No such file or directory" in str(ex):
```

Comment:
If you want to check for a tool and report with an error message if it is missing, use `shutil.which` first, before `subprocess.run`: https://docs.python.org/3/library/shutil.html#shutil.which.

Exception handling should generally use as specific an Exception subclass as possible, which might be OSError here: https://docs.python.org/3/library/subprocess.html#exceptions. You'll get a better error message if you check for the program before trying to run it and then not rely on the string output of a command that failed. String output can change at any time - between Python versions, across operating systems, etc.